--- a/easytier/Cargo.toml
+++ b/easytier/Cargo.toml
@@ -219,8 +219,6 @@
 ringbuf = "0.4.5"
 async-ringbuf = "0.3.1"
 
-service-manager = { git = "https://github.com/EasyTier/service-manager-rs.git", branch = "main" }
-
 zstd = { version = "0.13", optional = true }
 
 kcp-sys = { git = "https://github.com/EasyTier/kcp-sys", rev = "94964794caaed5d388463137da59b97499619e5f", optional = true }
--- a/easytier/src/lib.rs
+++ b/easytier/src/lib.rs
@@ -19,7 +19,6 @@
 pub mod peers;
 pub mod proto;
 pub mod rpc_service;
-pub mod service_manager;
 pub mod tunnel;
 pub mod utils;
 pub mod web_client;
