--- a/easytier/Cargo.toml
+++ b/easytier/Cargo.toml
@@ -230,10 +230,6 @@
 ] }
 
 # for http connector
-http_req = { git = "https://github.com/EasyTier/http_req.git", default-features = false, features = [
-    "rust-tls",
-] }
-
 # for dns connector
 hickory-resolver = "0.25.2"
 hickory-proto = "0.25.2"
--- a/easytier/src/connector/mod.rs
+++ b/easytier/src/connector/mod.rs
@@ -10,7 +10,6 @@
     },
     utils::BoxExt,
 };
-use http_connector::HttpTunnelConnector;
 use rand::seq::SliceRandom;
 
 pub mod direct;
@@ -19,7 +18,6 @@
 pub mod udp_hole_punch;
 
 pub mod dns_connector;
-pub mod http_connector;
 
 pub(crate) fn should_try_p2p_with_peer(
     feature_flag: Option<&PeerFeatureFlag>,
@@ -280,9 +278,6 @@
         }
         #[cfg(unix)]
         TunnelScheme::Unix => tunnel::unix::UnixSocketTunnelConnector::new(url).boxed(),
-        TunnelScheme::Http | TunnelScheme::Https => {
-            HttpTunnelConnector::new(url, global_ctx.clone()).boxed()
-        }
         TunnelScheme::Ring => RingTunnelConnector::new(url).boxed(),
         TunnelScheme::Txt | TunnelScheme::Srv => {
             if url.host_str().is_none() {
--- a/easytier/src/connector/manual.rs
+++ b/easytier/src/connector/manual.rs
@@ -88,7 +88,7 @@
     fn reconnect_timeout(dead_url: &url::Url) -> Duration {
         let use_long_timeout = matches_scheme!(
             dead_url,
-            TunnelScheme::Http | TunnelScheme::Https | TunnelScheme::Txt | TunnelScheme::Srv
+            TunnelScheme::Txt | TunnelScheme::Srv
         ) || matches!(dead_url.scheme(), "ws" | "wss");
 
         Duration::from_secs(if use_long_timeout { 20 } else { 2 })
--- a/easytier/src/tunnel/mod.rs
+++ b/easytier/src/tunnel/mod.rs
@@ -350,8 +350,6 @@
     #[cfg(unix)]
     Unix,
     // Only for connector
-    Http,
-    Https,
     Ring,
     Txt,
     Srv,
